### PR TITLE
update entity models to leverage non identity column based PKs

### PR DESCRIPTION
**Pull request by kpaul**
will follow up with the group service changes to take advantage of the new data model in a different PR

### CODE REVIEW

File: Fabric.Authorization.Persistence.SqlServer/Extensions/ModelBuilderExtensions.cs [hunk 1]
```diff
@@ -130,6 +135,10 @@ public static void ConfigureRole(this ModelBuilder modelBuilder)
             modelBuilder.Entity<Role>(entity =>
             {
                 entity.ToTable("Roles");
+
+                entity.Property(e => e.Id)
+                    .UseSqlServerIdentityColumn();
```
Comment (bsmithvt) [inline Fabric.Authorization.Persistence.SqlServer/Extensions/ModelBuilderExtensions.cs]: Should this have IsUnique?

File: Fabric.Authorization.Persistence.SqlServer/Extensions/ModelBuilderExtensions.cs [hunk 2]
```diff
@@ -184,6 +195,15 @@ public static void ConfigureGroup(this ModelBuilder modelBuilder)
...
                 entity.ToTable("Groups");
 
+                entity.HasKey(g => g.Name);
+
+                entity.HasIndex(e => e.Id)
+                    .IsUnique()
+                    .ForSqlServerIsClustered();
+
+                entity.Property(e => e.Id)
+                    .UseSqlServerIdentityColumn();
```
Comment (bsmithvt) [inline Fabric.Authorization.Persistence.SqlServer/Extensions/ModelBuilderExtensions.cs]: IsUnique?

File: Fabric.Authorization.Persistence.SqlServer/Extensions/ModelBuilderExtensions.cs (hunk 1 above)
Comment (kpaul) [reply Fabric.Authorization.Persistence.SqlServer/Extensions/ModelBuilderExtensions.cs]: I have the unique property on the index setup

File: Fabric.Authorization.Persistence.SqlServer/Extensions/ModelBuilderExtensions.cs (hunk 2 above)
Comment (kpaul) [reply Fabric.Authorization.Persistence.SqlServer/Extensions/ModelBuilderExtensions.cs]: same answer as above

**Pull request merged**